packages/es-components/src/components/containers/tabPanels/TabPanel.js

```diff
@@ -25,55 +25,41 @@ const TabFormatter = styled.div`
 const TabContent = styled.div`
   margin-top: -1px;
   background-color: ${props => props.theme.colors.white};
-  overflow: auto;
   border-top: 1px solid ${props => props.theme.colors.gray4};
 `;
 
 class TabPanel extends React.Component {
   constructor(props) {
     super(props);
-    const child = Array.isArray(this.props.children)
-      ? props.children[0]
-      : props.children;
-    this.state = {
-      value: child.props.name,
-      currentContent: child.props.children
-    };
     this.tabChanged = this.tabChanged.bind(this);
   }
 
-  componentDidUpdate(prevProp, prevState) {
-    if (this.state.value !== prevState.value) {
-      this.props.tabChanged(this.state.value);
-    }
-  }
-
-  tabChanged(name, child) {
-    this.setState({
-      value: name,
-      currentContent: child
-    });
+  tabChanged(header, parentTabChanged) {
+    parentTabChanged(header);
   }
 
   render() {
-    const { children } = this.props;
-    const elements = React.Children.map(children, (child, i) => {
-      const isSelected = child.props.name.key
-        ? child.props.name.key === this.state.value.key
-        : child.props.name === this.state.value;
-      return React.cloneElement(child, {
-        selected: isSelected,
-        action: this.tabChanged
-      });
-    });
-
+    const { children, selectedKey, tabChanged } = this.props;
+    let displayIndex = findIndex(children, (child) => 
+      child.props.header.key
+        ? child.props.header.key === selectedKey
+        : child.props.header === selectedKey
+      );
+      if(displayIndex < 0){
+        displayIndex = 0;
+      }
+    const elements = React.Children.map(children, (child, i) => React.cloneElement(child, {
+        selected: i === displayIndex,
+        action: (header) => {this.tabChanged(header, tabChanged)}
+    }));
```

Comment:
We aren't using state anymore. Is there a reason to keep `TabPanel` as a class rather than a function?